app/Http/Livewire/ExchangeTable.php

```diff
@@ -20,6 +37,20 @@ public function render(): View
 
     public function getExchanges(): Collection
     {
-        return Exchange::orderBy('name')->get();
+        return Exchange::orderBy('name')
+            ->filterByType($this->type)
+            ->filterByPair($this->pair)
+            ->get();
+    }
+
+    public function setFilter(string $param, string $value): void
+    {
+        if ($param === 'type') {
+            $this->type = $value;
+        }
```

Comment:
I can also use $this->{$param} = $type but that creates a rector cs-fixer warning, also is safer this way

Reply:
You could also validate the param beforehand if you wanted to do the `->{$param}` (assuming you can fix the warning)

```suggestion
    public function setFilter(string $param, string $value): void
    {
        if (! in_array($param, ['type', 'pair'])) {
            return;
        }
        
        $this->{$param} = $type;
    }
```

Reply:
What was the warning?

Reply:
@alexbarnsley `Variable property access on $this(App\Http\Livewire\ExchangeTable)`, tried your suggestion but does not handle the warning